Library/OcBootManagementLib/BootEntryManagement.c

```diff
@@ -714,12 +714,25 @@ InternalAddBootEntryFromCustomEntry (
     DEBUG_INFO,
     "OCB: Adding custom entry %s (%a|B:%d) -> %a\n",
     BootEntry->Name,
-    CustomEntry->SystemAction != NULL ? "action" : (CustomEntry->Tool ? "tool" : "os"),
+    CustomEntry->ExternalSystemAction != NULL ? "ext-action" : (CustomEntry->SystemAction != NULL ? "action" : (CustomEntry->Tool ? "tool" : "os")),
```

Comment:
Does this need icon update in OcBinaryData?

Reply:
Flavor still needs to be set like for SystemAction, currently OC_FLAVOUR_WINDOWS is used for all entries at the moment. Shouldn't need any icon updates.